index.js

```diff
@@ -22,7 +22,20 @@ function onexit () {
   exiting = true
 
   process.removeListener('beforeExit', onexit)
-  Promise.allSettled(handlers.map(fn => fn())).then(done, done)
+
+  const order = []
+
+  for (const h of handlers.sort((a, b) => b.position - a.position)) {
+    if (!order.length || order[order.length - 1][0].position !== h.position) order.push([])
+    order[order.length - 1].push(h)
+  }
+
+  loop()
+
+  function loop () {
+    if (!order.length) return done()
+    Promise.allSettled(order.pop().map(h => h.fn())).then(loop, loop)
+  }
```

Comment:
wow, beautiful!

Reply:
hahaha, your code! I wanted to left a comment saying that this is not lukks's code